src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,133 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.utils.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    public static final String DELETING_QUERY = "DELETE FROM books WHERE id=?";
+    private static final String CREATION_QUERY = "INSERT INTO books (title, price) VALUES (?,?)";
+    private static final String SEARCHING_QUERY = "SELECT * FROM books WHERE id=?";
+    private static final String SEARCHING_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATING_QUERY = "UPDATE books SET title=?, price=? WHERE id=?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(CREATION_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows != 1) {
+                throw new RuntimeException(
```

Comment:
you should create your custom `DataProcessingException` and throw it where applicable

**make sure to fix it in all places**